lgc/builder/YCbCrConvertHandler.cpp

```diff
@@ -0,0 +1,929 @@
+
+  return m_builder->CreateImageSampleGather(ycbcrInfo->resultTy, ycbcrInfo->dim, ycbcrInfo->flags, coords,
+                                            ycbcrInfo->imageDesc, ycbcrInfo->samplerDesc, ycbcrInfo->address,
+                                            ycbcrInfo->instNameStr, ycbcrInfo->isSample);
+}
+
+// =====================================================================================================================
+// YCbCrConvertHandler
+//
+// @param builder : ImageBuilder instance
+// @param ycbcrMetaData : YCbCr conversion metadata
+// @param ycbcrSampleInfo : YCbCr sample information
+// @param gfxIp : The GfxIp Version
+YCbCrConvertHandler::YCbCrConvertHandler(ImageBuilder *builder, SamplerYCbCrConversionMetaData &ycbcrMetaData,
+                                         YCbCrSampleInfo *ycbcrSampleInfo, GfxIpVersion *gfxIp) {
+  Register(builder);
+  setGfxIpVersion(gfxIp);
+  setConversionMetaData(ycbcrMetaData);
+  setYCbCrSampleInfo(ycbcrSampleInfo);
+  genSamplerDescChroma();
+  genImgDescChroma();
+}
+
+// =====================================================================================================================
+// Set YCbCr conversion meta data
+//
+// @param ycbcrMetaData : YCbCr conversion meta data
+void YCbCrConvertHandler::setConversionMetaData(SamplerYCbCrConversionMetaData &ycbcrMetaData) {
+  m_planeCount = ycbcrMetaData.word1.planes;
+  m_subsampledX = ycbcrMetaData.word1.xSubSampled;
+  m_subsampledY = ycbcrMetaData.word1.ySubSampled;
+  m_forceExplicitReconstruct = ycbcrMetaData.word0.forceExplicitReconstruct;
+  m_bitCountX = ycbcrMetaData.word2.bitCounts.xBitCount;
+  m_bitCountY = ycbcrMetaData.word2.bitCounts.yBitCount;
+  m_bitCountZ = ycbcrMetaData.word2.bitCounts.zBitCount;
+  m_bitCountW = ycbcrMetaData.word2.bitCounts.wBitCount;
+  m_bpp = (m_planeCount == 1) ? (m_bitCountX + m_bitCountY + m_bitCountZ + m_bitCountW) : m_bitCountX;
+  m_swizzleR = ycbcrMetaData.word0.componentMapping.swizzleR;
+  m_swizzleG = ycbcrMetaData.word0.componentMapping.swizzleG;
+  m_swizzleB = ycbcrMetaData.word0.componentMapping.swizzleB;
+  m_swizzleA = ycbcrMetaData.word0.componentMapping.swizzleA;
+  m_lumaFilter = static_cast<SamplerFilter>(ycbcrMetaData.word1.lumaFilter);
+  m_chromaFilter = static_cast<SamplerFilter>(ycbcrMetaData.word1.chromaFilter);
+  m_chromaOffsetX = static_cast<ChromaLocation>(ycbcrMetaData.word1.xChromaOffset);
+  m_chromaOffsetY = static_cast<ChromaLocation>(ycbcrMetaData.word1.yChromaOffset);
+  m_ycbcrRange = static_cast<SamplerYCbCrRange>(ycbcrMetaData.word0.yCbCrRange);
+  m_ycbcrModel = static_cast<SamplerYCbCrModelConversion>(ycbcrMetaData.word0.yCbCrModel);
+  m_tileOptimal = ycbcrMetaData.word1.tileOptimal;
+  m_bits[0] = ycbcrMetaData.word0.bitDepth.channelBitsR;
+  m_bits[1] = ycbcrMetaData.word0.bitDepth.channelBitsG;
+  m_bits[2] = ycbcrMetaData.word0.bitDepth.channelBitsB;
+  m_metaData = &ycbcrMetaData;
```

Comment:
There is a bit of an inconsistency here: most parts of the rest of the code refer to the member variables that are explicitly extracted here, but some pieces of code instead refer to the captured pointer m_metaData.

Could you please make this consistent? Either refer via m_metaData everywhere (with the exception of m_bpp, which is computed specially), or extract everything and remove m_metaData?

Reply:
Oh, yes, i'd prefer just refer via m_metaData everywhere except m_bpp, thanks.